lib/http/response.rb

```diff
@@ -138,8 +138,8 @@ def content_type
     def_delegator :content_type, :charset
 
     def cookies
-      @cookies ||= headers.each_with_object CookieJar.new do |(k, v), jar|
-        jar.parse(v, uri) if k == Headers::SET_COOKIE
+      @cookies ||= headers[Headers::SET_COOKIE].each_with_object CookieJar.new do |v, jar|
+        jar.parse(v, uri)
```

Comment:
`each_with_object` doesn't work because it's returning a string. How about something like:

```suggestion
      @cookies ||= CookieJar.new do |jar|
          jar.parse(headers[Headers::SET_COOKIE], url) if headers.has_key?(Headers::SET_COOKIE)
      end
```